.github/workflows/publish-package.yml

```diff
@@ -1,13 +1,12 @@
 name: Publish Package to GitHub
 on:
-  push:
-    branches: ['main']
-  pull_request:
-    branches: ['main']
+  workflow_dispatch:
+    branches:
+      - main
```

Comment:
Shouldn't this be rather a release: published trigger? You probably do not want to release every push to main to NPM directly.

Reply:
Thats actually what i wanted to introduce. Before it always released the merge/push on main - now i wanted to introduce only the manual trigger (if needed) but based on main branch and no other

Reply:
Wouldn't it be better to only do this on release creation rather than a manual trigger? Either that or at the same time create the proper git tag.